### PR TITLE
Added pagination, sorting and swagger

### PR DESCRIPTION
**Pull request by artemklishch**

### CODE REVIEW

File: src/main/java/org/example/intro/service/impl/BookServiceImpl.java [hunk 1]
```diff
@@ -49,15 +47,14 @@ public void delete(Long id) {
     }
 
     @Override
-    public List<BookDto> search(BookSearchParametersDto params, int page) {
-        Pageable pageable = PageRequest.of(page, ITEMS_PER_PAGE);
+    public List<BookDto> search(BookSearchParametersDto params, Pageable pageable) {
```
Comment (Elena-Bruyako) [inline src/main/java/org/example/intro/service/impl/BookServiceImpl.java:48]: Have you done the HW to implement search endpoint?

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Let's discuss

File: src/main/java/org/example/intro/service/impl/BookServiceImpl.java (hunk 1 above)
Comment (artemklishch) [reply src/main/java/org/example/intro/service/impl/BookServiceImpl.java:48]: This HW was related to "findAll" controller method.

But i decided to improve this method as well. I made this controller method in the previous PR. But here after watching new video lesson i thought to improve this method. I removed the path variable, where i put the page number before and updated like with normal fetching of all books - with using Pageable class.

Comment (Elena-Bruyako) [reply src/main/java/org/example/intro/service/impl/BookServiceImpl.java:48]: It's better to implement search endpoint according to the **Add ability to search books** HW (see Spring Boot / Criteria Query in Spring Boot section)

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
Answered you

File: src/main/java/org/example/intro/service/impl/BookServiceImpl.java (hunk 1 above)
Comment (artemklishch) [reply src/main/java/org/example/intro/service/impl/BookServiceImpl.java:48]: it seems i don't understand what to make here.
could you please elaborate what issues here?

Comment (Elena-Bruyako) [reply src/main/java/org/example/intro/service/impl/BookServiceImpl.java:48]: Screenshot 2024-07-24 at 13 14 37
Please, check this module and implementation of search endpoint. I didn't find any of specifications in your code

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
answered you

File: src/main/java/org/example/intro/service/impl/BookServiceImpl.java (hunk 1 above)
Comment (artemklishch) [reply src/main/java/org/example/intro/service/impl/BookServiceImpl.java:48]: ++, Thank you! Updated. I merged by mistake the PR where i wanted to make the "search" method. and made changes here - i didn't know how to restore that PR.

**Review by okuzan — APPROVED** [review]

**Pull request merged**